backend/src/repositories/influx-metrics-provider.repository.ts

```diff
@@ -27,19 +28,26 @@ export class InfluxMetricsProvider implements IMetricsProviderRepository {
     }
   }
 
-  async flush() {
+  flush(): Promise<void> {
     if (this.metrics.length === 0) {
-      return;
+      return Promise.resolve();
     }
     const metrics = this.metrics.join('\n');
-    const response = await fetch(`${this.influxApiUrl}/write`, {
-      method: 'POST',
-      body: this.metrics.join('\n'),
-      headers: {
-        Authorization: `Token ${this.influxApiToken}`,
-      },
-    });
-    await response.body?.cancel();
-    console.log(metrics);
+    this.metrics = [];
+    const send = async () => {
+      const response = await fetch(`${this.influxApiUrl}/write`, {
+        method: 'POST',
+        body: metrics,
+        headers: {
+          Authorization: `Token ${this.influxApiToken}`,
+        },
+      });
+      await response.body?.cancel();
+      if (response.status !== 204) {
+        throw new Error(`InfluxDB write failed with status ${response.status}`);
+      }
+    };
+
+    return send();
```

Comment:
Not sure if we need to define this method and then call it right away. We can just remove the method wrapper in that case.

Reply:
The reasoning for this is to avoid race conditions. I want this method to return a promise for the actual flush, however I want the initial part of the function, namely the join and then clear of the metrics array to run synchronously. So we run that part, then run the next part async and return the promise for the send to influx.